### PR TITLE
Add tz function

### PR DESCRIPTION
**Pull request by JuanMGN**
# Description

Add tz function according specification
https://www.w3.org/TR/sparql11-query/#func-tz

## Type of change

Make sure you add the correct label for the PR:

- `enhacement` if this PR adds a new feature
- `bug` if it fixes a bug
- `documentation` if it adds docs
- `breaking-change` if this PR introduces a breaking change
- `dependency-updates` if it updates dependencies

### CODE REVIEW

**Comment by puremachinery** [comment]
LGTM, though I won't try to Approve since I'm still new to this codebase and prefer to have Pepe or Patricio also take a look before it gets auto-merged. :)

File: modules/engine/src/main/scala/com/gsk/kg/engine/functions/FuncDates.scala [hunk 1]
```diff
@@ -170,4 +148,97 @@ object FuncDates {
...
+
+  /** Get timezone of a datetime input in multiples formats
+    * @param col
+    * @return timezone formatted
+    *         usage
+    *         getTimeZoneComponents(2020-12-09T01:50:24.888Z) = Z
+    *         getTimeZoneComponents(2011-01-10T14:45:13.815-05:29) = -05:29
+    *         getTimeZoneComponents(2011-01-10T14:45:13.815+05:09) = 05:29
+    */
+  private def getTimeZoneComponents(col: Column): Column = {
```
Comment (pepegar) [inline modules/engine/src/main/scala/com/gsk/kg/engine/functions/FuncDates.scala:160]: 👍🏻

File: modules/engine/src/main/scala/com/gsk/kg/engine/functions/FuncDates.scala [hunk 2]
```diff
@@ -170,4 +148,97 @@ object FuncDates {
...
+    * @return timezone formatted
+    *         usage
+    *         buildTimeZone(-05:29) = "\"-PT5H29M\"^^xsd:dateTime"
+    *         buildTimeZone(05:00)  = "\"PT5H\"^^xsd:dateTime"
+    */
+  private def buildTimeZone(
+      timeZone: Column,
+      PosSignOpt: Option[Int],
+      PosHours: Int,
+      PosMinutes: Int
```
Comment (pepegar) [inline modules/engine/src/main/scala/com/gsk/kg/engine/functions/FuncDates.scala:220]: I'd convert these to lowercase.  They'd only be constants if they were plain variables, not arguments.

**Review by pepegar — APPROVED** [review]
LGTM

**Pull request merged**